src/algebra/lie/basic.lean

```diff
@@ -891,6 +926,84 @@ end lie_span
 
 end lattice_structure
 
+section lie_ideal_operations
+
+/-- Given a Lie module `M` over a Lie algebra `L`, the set of Lie ideals of `L` acts on the set
+of submodules of `M`. -/
+instance : has_bracket (lie_ideal R L) (lie_submodule R L M) :=
+⟨λ I N, lie_submodule.lie_span R L { m | ∃ (x : I) (n : N), ⁅(x : L), (n : M)⁆ = m }⟩
+
+lemma lie_apply :
+  ⁅I, N⁆ = lie_submodule.lie_span R L { m | ∃ (x : I) (n : N), ⁅(x : L), (n : M)⁆ = m } := rfl
+
+lemma lie_mem_lie (x : I) (m : N) : ⁅(x : L), (m : M)⁆ ∈ ⁅I, N⁆ :=
+by { rw lie_apply, apply subset_lie_span, use [x, m], }
+
+lemma lie_le : ⁅I, N⁆ ≤ N :=
```

Comment:
I think you'll also want `[I, J] \le I` when you specialise to ideals, so these should probably be called `lie_le_right` and `lie_le_left`.
```suggestion
lemma lie_le_right : ⁅I, N⁆ ≤ N :=
```

Reply:
Great point!